lldb-eval/eval.cc

```diff
@@ -1184,13 +1244,13 @@ Value Interpreter::EvaluateBinarySubAssign(Value lhs, Value rhs) {
 
   if (lhs.IsPointer()) {
     assert(rhs.IsInteger() && "invalid ast: rhs must be an integer");
-    ret = EvaluateBinarySubtraction(lhs, rhs);
+    ret = EvaluateBinarySubtraction(lhs, rhs, rhs.type());
```

Comment:
Shouldn't the result type here be a pointer? Is there a test for pointer-integer subtract-assign operation?

Reply:
Oops, you're right, it should be `lhs.type()`.
We have tests for `pointer -= integer`, but the third argument in `EvaluateBinarySubtraction` is only used for `ptrdiff_t`, so in this case it will be ignored.